### DISCUSSION TITLE
Passing date object from server component triggers warning

### DISCUSSION THREAD

**Discussion by kieranm** [amannn/next-intl#162] [Q&A]
Hi all,
I've set up the `NextIntlServerProvider` as directed by [the docs](https://next-intl-docs.vercel.app/docs/next-13/server-components)
```
const LocaleLayout: React.FC = (props) => {
  const { children, session, params } = props as RootLayoutProps
  const now = DateTime.utc()
  ...

  return (
    <html lang={params.locale}>
      <head />
      <body className="bg-background">
        <StrictMode>
          <NextIntlServerProvider
            locale={params.locale}
            timeZone={timeZone.effective}
            now={now.toJSDate()}
          >
              {children}
            </Providers>
        </StrictMode>
      </body>
    </html>
  )
}

export default RootLayout
```

However, this results in the following warning...
```
Warning: Only plain objects can be passed to Client Components from Server Components. Date objects are not supported.
  {locale: "fr", now: Date, timeZone: ...}
                      ^^^^
```

Is this a bug, or am I doing something wrong? Thanks!

**Comment by amannn** [comment]
Thanks for pointing this out, this is correct! Probably this will be resolved upstream in the future: https://github.com/facebook/react/issues/25687

Maybe, at least for the period until RSC becomes stable, we could support passing a timestamp number as `now`, I've added this to my list in https://github.com/amannn/next-intl/pull/149.

Thanks again for the report!

**Reply by amannn** [reply]
The latest iteration of the RSC integration no longer has this issue: https://github.com/amannn/next-intl/pull/149#issuecomment-1376135315
